### PR TITLE
Port onboarding on Android

### PR DESCRIPTION
**Pull request by yrliou**
Fix https://github.com/brave/brave-browser/issues/6548

Screenshot_20191023-112513
Screenshot_20191023-112539
Screenshot_20191023-112603

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -216,12 +216,52 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      <message name="IDS_BRAVE_ADS_ONBOARDING_TITLE" desc="Title for brave ads page">Brave will show your first ad in</message>
+      <message name="IDS_BRAVE_ADS_ONBOARDING_TEXT" desc="Text for brave ads page">To see ads from Brave Rewards, make sure Brave is allowed to show you notifications.</message>
+
+      <message name="IDS_THIS_IS_YOUR_FIRST_AD" desc="Text brave rewards notification.">This is your first Brave ad. Tap to learn more.</message>
+
+      <message name="IDS_TURN_ON" desc="Text for turn on button">Turn On</message>
+
+      <message name="IDS_DO_NOT_AGREE" desc="Text for do not agree button">Do not agree</message>
+
+      <message name="IDS_FINISH" desc="Text for finish button">Finish</message>
```
Comment (SergeyZhukovsky) [inline android/java/strings/android_brave_strings.grd]: Could we make them the same style as strings above?
<message ...>
  ....
</message>
That style comes from Chromium, better to be consistent with that.

Comment (yrliou) [reply android/java/strings/android_brave_strings.grd]: addressed in https://github.com/brave/brave-core/pull/3768/commits/251335fedf2f7c9444eb56d9044e20f2f6e81631

**Comment by yrliou** [comment]
Mark CI/skip on other platforms expect for android because this PR only touches files under android folder.

File: android/java/org/chromium/chrome/browser/notifications/BraveAdsNotificationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+            fontScale = Math.min(fontScale, FONT_SCALE_LARGE);
+            paddingScale = (FONT_SCALE_LARGE - fontScale) / (FONT_SCALE_LARGE - 1.0f);
+        }
+        return dpToPx(metrics, paddingScale * MAX_SCALABLE_PADDING_DP);
+    }
+
+    /**
+     * Converts a px value to a dp value.
+     */
+    private static int pxToDp(float value, DisplayMetrics metrics) {
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/notifications/BraveAdsNotificationBuilder.java]: I think we should have this function in a separate util class as we can use it at different places.

Comment (yrliou) [reply android/java/org/chromium/chrome/browser/notifications/BraveAdsNotificationBuilder.java]: Thanks for the review!
This is from current production code, how about we do this in a follow up?

Comment (yrliou) [reply android/java/org/chromium/chrome/browser/notifications/BraveAdsNotificationBuilder.java]: As discussed in DM, we'll merge this one as is first.

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/notifications/BraveAdsNotificationBuilder.java]: @yrliou yeah it should be fine. I am updating onboarding for latest release. So we need to push that change as well.

**Pull request merged**